src/components/services/OctoPi.vue

```diff
@@ -0,0 +1,165 @@
+    <template #content>
+      <p class="title is-4">{{ item.name }}</p>
+      <p class="subtitle is-6">
+        <template v-if="item.subtitle && printerstatus == null">
+          {{ item.subtitle }}
+        </template>
+        <template v-else-if="api" && printerstatus !="null">
+          Status: {{ printerstatus }}
+        </template>
+        <template v-if="printerstatus == 'Printing'"
+          >|| Progress: {{ printerprogress }}% <br />
+          Elapsed: {{ printtime }} || Left: {{ printtimeleft }}
+        </template>
+      </p>
+    </template>
+  </Generic>
+</template>
+
+<script>
+import service from "@/mixins/service.js";
+import Generic from "./Generic.vue";
+
+export default {
+  name: "OctoPi",
+  mixins: [service],
+  props: {
+    item: Object,
+  },
+  components: {
+    Generic,
+  },
+  data: () => ({
+    api: {
+      code: true,
+      state: "",
+      job: {
+        file: {
+          display: "",
+        },
+      },
+      progress: {
+        state: "",
+        completion: 0,
+        printTimeLeft: 0,
+        printTime: 0,
+      },
+    },
+  }),
+  computed: {
+    printerstatus: function () {
+      if (this.api && this.api.code != false) {
+        return this.api.state;
+      } else if (this.api.code != false) {
+        return "Offline";
+      }
+      return null;
+    },
+    printerprogress: function () {
+      if (this.api) {
+        return this.api.progress.completion.toFixed();
+      }
+      return 0;
+    },
+    printtime: function () {
+      if (this.api) {
+        var timestamp = this.api.progress.printTime;
+        var hours = Math.floor(timestamp / 60 / 60);
+        var minutes = Math.floor(timestamp / 60) - hours * 60;
+        var seconds = timestamp % 60;
+        if (hours < 10) {
+          hours = "0" + hours;
+        }
+        if (minutes < 10) {
+          minutes = "0" + minutes;
+        }
+        if (seconds < 10) {
+          seconds = "0" + seconds;
+        }
+        var formatted = hours + ":" + minutes;
+        return formatted;
+      }
+      return 0;
+    },
+    printtimeleft: function () {
+      if (this.api) {
+        var timestamp = this.api.progress.printTimeLeft;
+        var hours = Math.floor(timestamp / 60 / 60);
+        var minutes = Math.floor(timestamp / 60) - hours * 60;
+        var seconds = timestamp % 60;
+        if (hours < 10) {
+          hours = "0" + hours;
+        }
+        if (minutes < 10) {
+          minutes = "0" + minutes;
+        }
+        if (seconds < 10) {
+          seconds = "0" + seconds;
+        }
+        var formatted = hours + ":" + minutes;
+        return formatted;
+      }
+      return 0;
+    },
+  },
+  created() {
+    this.fetchStatus();
+  },
+  methods: {
+    fetchStatus: async function () {
+      if (this.item.apikey) {
+        this.api = await this.fetch(
+          "api/job?apikey=" + `${this.item.apikey}`
+        ).catch((e) => console.log(e), (this.api.code = false));

```

Comment:
Implement with a try-catch instead, and log the error with console.error. Can also simplify the string literal a bit:

```suggestion
        try {
          this.api = await this.fetch(`api/jov?apikey=${this.item.apikey}`);
        } catch (e) {
          console.error(e);
          this.api.code = false;
        }
```